base/reduce.jl

```diff
@@ -372,7 +428,7 @@ In the former case, the integers are widened to system word size and therefore
 the result is 128. In the latter case, no such widening happens and integer
 overflow results in -128.
 """
-sum(f::Callable, a) = mapreduce(promote_sys_size_add ∘ f, +, a)
+sum(f, a) = mapreduce(f, add_sum, a)
```

Comment:
Good catch: it looks like this Callable is not necessary for avoiding ambiguity any more. The same change should apply for `prod`.

Reply:
Fixed.